tests/installation/welcome.pm

```diff
@@ -18,36 +18,53 @@ use testapi;
 use utils 'ensure_fullscreen';
 
 sub run() {
-    my @welcome_tags = [qw(inst-welcome inst-welcome-confirm-self-update-server)];
-    ensure_fullscreen;
-    my $bootup_timeout = 500;
-    if (get_var("BETA")) {
-        assert_screen "inst-betawarning", $bootup_timeout;
-        send_key "ret";
-        assert_screen @welcome_tags, 10;
+    while (1) {
+        my @welcome_tags = ('inst-welcome-confirm-self-update-server', 'scc-invalid-url');
+        if (get_var('BETA')) {
+            push @welcome_tags, 'inst-betawarning';
+        }
+        else {
+            push @welcome_tags, 'inst-welcome';
+        }
+        ensure_fullscreen;
+        assert_screen \@welcome_tags, 500;
+        if (match_has_tag 'scc-invalid-url') {
+            die 'SCC reg URL is invalid' if !get_var('SCC_URL_VALID');
+            send_key 'alt-r';    # registration URL field
+            send_key_until_needlematch 'scc-invalid-url-deleted', 'backspace';
+            type_string get_var('SCC_URL_VALID');
+            wait_still_screen 2;
+            send_key 'alt-o';    # OK
```

Comment:
why not use wait_screen_change as on line 41? (without wait_still_screen) ..just for consistency :)